### PR TITLE
Update to release 0.2.9.2.3

### PR DESCRIPTION
**Pull request by zmanuel**
This release makes a lot of the special adaptions (patches and post-install fixes) obsolete, so they were removed.
Update the runtime to 23.08.
Switch to explicitly using sdl12-compat over native SDL1.2.
Note: This switch changes the default VSYNC behavior for some systems from WAIT to DO NOT WAIT. We have a menu item to customize that now in the screen setup.

### CODE REVIEW

**Comment by zmanuel** [comment]
Aww man, it works in our private repository. I hope the reason is something stupid simple.
Aha, but local builds with current tools also fail with the same errors. I suppose the CI environment is older and tolerates whatever this error is.

**Comment by zmanuel** [comment]
Previous log sniplet:
```
Removing files/include
Running appstream-compose
Processing application org.armagetronad.ArmagetronAdvanced
Saving icon /app/share/app-info/icons/flatpak/64x64/org.armagetronad.ArmagetronAdvanced.png
Saving AppStream /app/share/app-info/xmls/org.armagetronad.ArmagetronAdvanced.xml.gz
Done!
```
Now:
```
Removing files/include
Renaming org.armagetronad.ArmagetronAdvanced.appdata.xml to share/metainfo/org.armagetronad.ArmagetronAdvanced.metainfo.xml
Running appstreamcli compose
Saving screenshots in /srv/buildbot/worker/build-x86_64/build/.flatpak-builder/rofiles/rofiles-cQcJrk/files/share/app-info/media
FB: Running: appstreamcli compose --no-partial-urls --prefix=/ --origin=org.armagetronad.ArmagetronAdvanced --media-baseurl=https://dl.flathub.org/media --media-dir=/srv/buildbot/worker/build-x86_64/build/.flatpak-builder/rofiles/rofiles-cQcJrk/files/share/app-info/media --result-root=/srv/buildbot/worker/build-x86_64/build/.flatpak-builder/rofiles/rofiles-cQcJrk/files --data-dir=/srv/buildbot/worker/build-x86_64/build/.flatpak-builder/rofiles/rofiles-cQcJrk/files/share/app-info/xmls --icons-dir=/srv/buildbot/worker/build-x86_64/build/.flatpak-builder/rofiles/rofiles-cQcJrk/files/share/app-info/icons/flatpak '--components=org.armagetronad.ArmagetronAdvanced,org.armagetronad.ArmagetronAdvanced.desktop' /srv/buildbot/worker/build-x86_64/build/.flatpak-builder/rofiles/rofiles-cQcJrk/files
Only accepting components: org.armagetronad.ArmagetronAdvanced, org.armagetronad.ArmagetronAdvanced.desktop
Processing directory: /srv/buildbot/worker/build-x86_64/build/.flatpak-builder/rofiles/rofiles-cQcJrk/files
Composing metadata...
Run failed, some data was ignored.
Errors were raised during this compose run:
org.armagetronad.ArmagetronAdvanced
  E: icon-not-found
general
  E: filters-but-no-output
Refer to the generated issue report data for details on the individual problems.
Error: ERROR: appstreamcli compose failed: Child process exited with code 1
```

The line `Renaming org.armagetronad.ArmagetronAdvanced.appdata.xml to share/metainfo/org.armagetronad.ArmagetronAdvanced.metainfo.xml`
Is the first new thing. Apparently, the source is the legacy appstream metadata location.

**Comment by zmanuel** [comment]
AH YES STRAIGHTFORWARD, from the appstreamcli source:
```
{ "icon-not-found",
	  AS_ISSUE_SEVERITY_ERROR,
	  "The icon <em>{{icon_fname}}</em> was not found in the archive. This issue can have multiple reasons, "
	  "like the icon being in a wrong directory or not being available in a suitable size (at least 64x64px). "
 ```
 I suppose the extended text would have landed in the report the error message refers to, which is nowhere to be found, neither in the build bot output nor for the local build.
 
 We don't have a 64x64 icon.

File: armagetronad-appdata.patch [hunk 1]
```diff
@@ -1,33 +1,46 @@
...
+index 516025448..08be527af 100644
 --- a/desktop/armagetronad.appdata.xml.in
 +++ b/desktop/armagetronad.appdata.xml.in
-@@ -24,16 +24,24 @@
+@@ -4,7 +4,16 @@
+   
+   <name>@progtitle@</name>
+   <summary>3D Lightcycle Game</summary>
+-  
++  <developer id="tld.vendor">
```
Comment (AsciiWolf) [inline armagetronad-appdata.patch:10]: @zmanuel The id should be an actual id, not this example one.

Comment (AsciiWolf) [reply armagetronad-appdata.patch:10]: `org.armagetronad` should be the correct id.

Comment (zmanuel) [reply armagetronad-appdata.patch:10]: Ah, thanks for the correction. Sorry, as you can tell, I try to interact with these metadata systems as little as possible.

Comment (zmanuel) [reply armagetronad-appdata.patch:10]: OH just the tld.vendor thing! I though that was a fixed thing that had to be there so the whole section would be recognized. That makes more sense now.

**Pull request merged**